### PR TITLE
doc: openthread: add CSL feature information

### PR DESCRIPTION
**Pull request by edmont**
Add a summarized description of the CSL Thread 1.2 feature.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

### CODE REVIEW

**Review by MarekPorwisz — APPROVED** [review]

File: doc/nrf/ug_thread_supported_features.rst [hunk 1]
```diff
@@ -46,6 +46,16 @@ The Thread 1.2 Specification support has the following limitation:
 
 * Due to code size limitation, the combination of complete set of Thread 1.2 features with the Bluetooth® LE multiprotocol support is not possible for the nRF52833 DKs.
 
+Coordinated Sampled Listening (CSL)
+===================================
+
+Coordinated Sampled Listening defined in IEEE 802.15.4-2015 is introduced by Thread 1.2 Specification to provide low latency communication for Sleepy End Devices.
+Thread 1.2 are required to support synchronized CSL transmissions to children which require them, so called Synchronized Sleepy End Devices (SSEDs).
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_supported_features.rst:53]:
```suggestion
Thread 1.2 devices are required to support synchronized CSL transmissions to children which require them, known as Synchronized Sleepy End Devices (SSEDs).
```
Is devices the correct word here? Not 100% sure.

File: doc/nrf/ug_thread_supported_features.rst [hunk 2]
```diff
@@ -46,6 +46,16 @@ The Thread 1.2 Specification support has the following limitation:
...
+SSEDs are allowed to transmit frames normally at any time, but the routers should use the CSL transmission mechanism as long as the synchronization is maintained.
+This way an SSED is able to stay in a sleepy state more than 99% of the time and only turn on its radio periodically for a few hundreds of microseconds in order to receive frames from its parent.
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_supported_features.rst:55]:
```suggestion
This allows an SSED to stay in a sleepy state more than 99% of the time and only turn on its radio periodically for a few hundreds of microseconds in order to receive frames from its parent.
```

File: doc/nrf/ug_thread_supported_features.rst [hunk 3]
```diff
@@ -46,6 +46,16 @@ The Thread 1.2 Specification support has the following limitation:
...
+
+The most common use case for SSEDs is for developing low latency battery powered actuators, like window blinds.
```
Comment (wiba-nordic) [inline doc/nrf/ug_thread_supported_features.rst:57]:
```suggestion
The most common use case for SSEDs is for developing low latency battery-powered actuators, such as window blinds.
```

**Review by wiba-nordic — DISMISSED** [review]
A couple of small improvements I spotted.

**Comment by SebastianBoe** [comment]
@wiba-nordic : Your feedback has been addressed

**Review by greg-fer — APPROVED** [review]
Approving as the 2nd tech writer for Thread.

**Pull request merged**